src/software/ai/hl/stp/tactic/receiver/receiver_fsm.cpp

```diff
@@ -156,37 +158,29 @@ void ReceiverFSM::updateReceive(const Update& event)
 
 void ReceiverFSM::adjustReceive(const Update& event)
 {
-    auto ball      = event.common.world_ptr->ball();
-    auto robot_pos = event.common.robot.position();
-
-    if ((ball.position() - robot_pos).length() >
-        BALL_TO_FRONT_OF_ROBOT_DISTANCE_WHEN_DRIBBLING)
-    {
-        Point ball_receive_pos = ball.position();
-
-        if (ball.velocity().length() > MIN_PASS_START_SPEED)
-        {
-            ball_receive_pos = closestPoint(
-                robot_pos, Line(ball.position(), ball.position() + ball.velocity()));
-        }
-
-        Angle ball_receive_orientation = (ball.position() - robot_pos).orientation();
-
-        event.common.set_primitive(std::make_unique<MovePrimitive>(
-            event.common.robot, ball_receive_pos, ball_receive_orientation,
-            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
-            TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE,
-            TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
-            AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
-    }
+    auto face_ball_orientation =
+        (event.common.world_ptr->ball().position() - event.common.robot.position())
+            .orientation();
+
+    Point intercept_position =
+        findInterceptionPoint(event.common.robot, event.common.world_ptr->ball(),
+                              event.common.world_ptr->field()) +
+        Vector::createFromAngle(face_ball_orientation).normalize(0.05);
```

Comment:
why do we need this?

Reply:
I copied this over from the old interception code used in DribbleFSM. I think it has to do with #3061 #2859

Reply:
There's probably a better approach to this, I will revise the interception/dribble logic once I merge #3234 into new gameplay